dev/version.go

```diff
@@ -44,11 +54,64 @@ func (info *VersionInfo) String() string {
 	if info.Tag != "" {
 		return info.Tag
 	}
+
+	// use the same format as go: https://go.dev/doc/modules/version-numbers
+	nextVersion := info.NextVersion
+	if nextVersion == "" {
+		nextVersion = "v0.0.0"
+	}
 	if info.Commit != "" {
-		return info.Commit
+		return fmt.Sprintf("%s-%s-%s", nextVersion, pseudoversionTimestamp(time.Now()), info.Commit[:12])
 	}
 	if info.Dev != "" {
-		return "dev-" + info.Dev
+		return fmt.Sprintf("%s-%s-dev-%s", nextVersion, pseudoversionTimestamp(time.Now()), info.Dev[:12])
```

Comment:
This timestamp will end up being used as the cache key for nested execs, so I think it will result in more invalidations than are necessary.

Not the utter end of the world, just a local dev testing thing, but I do like the property of this being purely content hashed from the dir. Can we just use all zeros for the timestamp?

Reply:
Yeahhhh, bleh. Good call, fixed this to use all zeroes. All `-dev` versions are now stripped neatly :tada: So `vX.Y.Z-dev-123` is counted as compatible with `vX.Y.Z-dev-456`.